### PR TITLE
Common walls

### PR DESCRIPTION
**Pull request by bpark1327**
## Pull Request Description

For Attached/Multifamily buildings, model surfaces with `InteriorAdjacentTo` == `ExteriorAdjacentTo` as adiabatic, to accommodate, e.g., a wall between the unit's unconditioned space and the neighboring unit's same unconditioned space. 

Previously these surfaces were skipped, but this was somewhat unsatisfactory since A) there can be situations when you want to know that surface exists, B) this can make EnergyPlus unhappy if not enough surfaces are defined for a thermal zone, and/or C) secondary effects like thermal mass are not captured.

## Checklist

Not all may apply:

- [ ] EPvalidator.rb has been updated
- [x] Tests (and test files) have been updated
- [x] Documentation has been updated
- [x] `openstudio tasks.rb update_measures` has been run
- [x] No unexpected regression test changes on CI

### CODE REVIEW

File: tasks.rb
Comment (shorowit) [inline tasks.rb:1373]: A crawlspace wall is typically a FoundationWall, not (above-grade) Wall.

It'd be good to have have the file test multiple surfaces: adiabatic FoundationWall, adiabatic Wall, and adiabatic FrameFloor. Otherwise I can't tell if your code will work for those other surface types.

File: tasks.rb
Comment (shorowit) [inline tasks.rb:1385]: I don't agree with this change. It applies to all walls where `is_interior` is true, not just walls where InteriorAdjacentTo=ExteriorAdjacentTo.

File: HPXMLtoOpenStudio/measure.rb
Comment (shorowit) [inline HPXMLtoOpenStudio/measure.rb:384]: This feels unnecessary. Why would we allow InteriorAdjacentTo=living space and ExteriorAdjacentTo=living space, but not InteriorAdjacentTo=living space and ExteriorAdjacentTo=basement - conditioned? Or is there a nuance here I'm not appreciating?

File: HPXMLtoOpenStudio/measure.rb
Comment (shorowit) [inline HPXMLtoOpenStudio/measure.rb:3429]: 👍

File: workflow/tests/hpxml_translator_test.rb
Comment (shorowit) [reply workflow/tests/hpxml_translator_test.rb:655]: Add similar test for FrameFloor?

File: HPXMLtoOpenStudio/measure.rb
Comment (shorowit) [reply HPXMLtoOpenStudio/measure.rb:3429]: Actually, this would probably miss the case of a FrameFloor between living space and conditioned basement. How about adding a `is_adiabatic` method (similar to `is_interior` and `is_exterior` methods) to the HPXML class that checks for InteriorAdjacentTo=ExteriorAdjacentTo as well as this conditioned basement scenario?

File: workflow/tests/hpxml_translator_test.rb
Comment (bpark1327) [reply workflow/tests/hpxml_translator_test.rb:652]: Ah, you're right. I moved this test on the top of this loop because of `next unless wall.is_exterior` in line 663. I guess this will need to be replaced with an IF statement.

Comment (shorowit) [reply workflow/tests/hpxml_translator_test.rb:652]: Sounds right.

File: workflow/tests/hpxml_translator_test.rb
Comment (bpark1327) [reply workflow/tests/hpxml_translator_test.rb:654]: It was to draw your attention to how I proposed to test adiabatic surfaces below. It'll be removed.

File: HPXMLtoOpenStudio/measure.rb
Comment (bpark1327) [reply HPXMLtoOpenStudio/measure.rb:384]: My intention was to prevent users from describing interior partitions using RimJoists, Walls, FoundationWalls, or FrameFloors. I wanted these constructions to not have conditioned space types (i.e. living space and basement - conditioned) as `interior_adjacent_to` and `exterior_adjacent_to` at the same time. 
However, as you pointed out, this wasn't doing what I intended 🤦 
Do you still think it's unnecessary to correct this part and prevent users from describing interior partitions?

Comment (shorowit) [reply HPXMLtoOpenStudio/measure.rb:393]: Hmm. I tend to feel that this is not necessary. I am pulling in the latest master and will remove this check.

File: HPXMLtoOpenStudio/resources/hpxml.rb
Comment (shorowit) [inline HPXMLtoOpenStudio/resources/hpxml.rb:1557]: These methods will either return true or nil, as opposed to true or false. Probably still works properly as written but I will address this.

**Review by shorowit — APPROVED** [review]
This looks good. Will wait on the CI and double-check results before merging.

**Comment by shorowit** [comment]
No CI diffs. 👍

**Comment by shorowit** [comment]
Thanks @bpark1327!

**Pull request merged**
